src/memcached_cache.cpp

```diff
@@ -248,14 +249,38 @@ ImpuStore::Impu* MemcachedCache::get_impu_for_impu_gr(const std::string& impu,
 
   if (!data)
   {
+    std::vector<std::promise<ImpuStore::Impu*>*> promises;
+
     for (ImpuStore* remote_store : _remote_stores)
     {
-      data = remote_store->get_impu(impu, trail);
+      std::promise<ImpuStore::Impu*>* promise = new std::promise<ImpuStore::Impu*>();
+      promises.push_back(promise);
 
-      if (data)
+      _thread_pool.add_work([promise, remote_store, impu, trail]()->void
       {
-        break;
+        promise->set_value(remote_store->get_impu(impu, trail));
+      });
+    }
+
+    for (std::promise<ImpuStore::Impu*>* promise : promises)
+    {
+      std::future<ImpuStore::Impu*> future = promise->get_future();
```

Comment:
Possibly worth a comment stating that this will wait until the future has a valid result, as i had to look it up to check. And that the timeout for that wait will be <thread queue time> + <remote_store get_impu timeout>

Reply:
That's just how futures work - I don't think we need to comment that.